plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3850,7 +3850,7 @@ public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
         // we need to chop off any "$partitions" and similar suffixes from table name while querying the metastore for the Table object
         TableNameSplitResult tableNameSplit = splitTableName(tableName.getTableName());
         Optional<Table> table = metastore.getTable(tableName.getSchemaName(), tableNameSplit.getBaseTableName());
-        Optional<CatalogSchemaTableName> catalogSchemaTableName = tableRedirectionsProvider.redirectTable(session, table);
+        Optional<CatalogSchemaTableName> catalogSchemaTableName = tableRedirectionsProvider.redirectTable(session, tableName, table);
```

Comment:
We should probably create a new `SchemaTableName` from the `tableNameSplit`. Otherwise it is going to be a bit confusing if we chop off meta-table suffixes like `$partitions`. For example if I said `SELECT * FROM db.foo$partitions`, and `foo` exists, then the `Table` argument we pass to `HiveTableRedirectionsProvider#redirectTable()` will have a name of `db.foo`, but the `SchemaTableName` argument will have `db.foo$partitions`. It would be better to be consistent about the suffix removal.

Reply:
Agree with your suggestion and have made this change.